### PR TITLE
Add config flow to Synology DSM

### PR DESCRIPTION
**Pull request by Quentame**
## Proposed change

Follows parrent PR home-assistant/core#32704

And so following breaking changes "Integration has been rename from `synologydsm` to `synology_dsm`" from parent PR home-assistant/core#32704

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: home-assistant/core#32704
- Link to parent pull request in the Brands repository: home-assistant/brands/pull/876
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/synology_dsm.markdown [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+ha_category:
+  - System Monitor
+ha_release: 0.32
+ha_iot_class: Local Polling
+ha_domain: synology_dsm
+---
+
+The `synology_dsm` sensor platform allows getting various statistics from your [Synology NAS](https://www.synology.com).
+
+## Setup the integration
```
Comment (frenck) [inline source/_integrations/synology_dsm.markdown]:
```suggestion
## Configuration
```

File: source/_integrations/synology_dsm.markdown [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+There is two ways to integrate Synology DSM in Home Assistant.
```
Comment (frenck) [inline source/_integrations/synology_dsm.markdown]:
```suggestion
There are two ways to integrate Synology DSM in Home Assistant.
```

File: source/_integrations/synology_dsm.markdown [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+<div class='note warning'>
+
+This sensor will wake up your Synology NAS if it's in hibernation mode.
+
+</div>
+
+## Separate User Configuration
+
+Due to the nature of the Synology DSM API it is required to grant the user admin rights. This is related to the fact that utilization information is stored in the core module.
```
Comment (frenck) [inline source/_integrations/synology_dsm.markdown]:
```suggestion
Due to the nature of the Synology DSM API, it is required to grant the user admin rights. This is related to the fact that utilization information is stored in the core module.
```

File: source/_integrations/synology_dsm.markdown [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+When creating the user it is possible to deny access to all locations and applications. By doing this the user will not be able to login to the web interface or view any of the files on the Synology NAS. It is still able to read the utilization and storage information using the API.
```
Comment (frenck) [inline source/_integrations/synology_dsm.markdown]:
```suggestion
When creating the user, it is possible to deny access to all locations and applications. By doing this, the user will not be able to login to the web interface or view any of the files on the Synology NAS. It is still able to read the utilization and storage information using the API.
```

File: source/_integrations/synology_dsm.markdown [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+<div class='note warning'>
+
+Using two-factor authentication is not supported. Please use a strong randomly generated password.
```
Comment (frenck) [inline source/_integrations/synology_dsm.markdown]:
```suggestion
Using two-factor authentication is not supported. Please use a strong, randomly generated password.
```

File: source/_integrations/synology_dsm.markdown [hunk 6]
```diff
@@ -0,0 +1,128 @@
+---
+title: Synology DSM
+description: Instructions on how to integrate the Synology DSM sensor within Home Assistant.
+ha_category:
+  - System Monitor
+ha_release: 0.32
+ha_iot_class: Local Polling
+ha_domain: synology_dsm
```
Comment (frenck) [inline source/_integrations/synology_dsm.markdown:8]:
```suggestion
ha_domain: synologydsm
```

**Review by frenck — CHANGES_REQUESTED** [review]
A redirect from the old page needs to be added to the redirects file.

**Comment by Quentame** [comment]
Redirect seems to work well (tested on the netlify/home-assistant-docs/deploy-preview build)

**Review by frenck — APPROVED** [review]

**Pull request merged**